types/parser/type_parser.go

```diff
@@ -122,7 +122,7 @@ func (t *typename) Capture(values []string) error {
 }
 
 type nonParamType struct {
-	TypeName    typename `parser:"@(IntType | Boolean | FPType | Temporal | BinaryType)"`
+	TypeName    typename `parser:"@(IntType | Boolean | FPType | Temporal | BinaryType | UserDefinedType)"`
```

Comment:
`RetType()` probably needs to be updated to handle a user-defined type scenario. I'm pretty sure `SimpleTypeNameToType` will not currently return a user-defined type, it'll error out.